app/models/user.rb

```diff
@@ -916,12 +916,6 @@ def can_edit_registration?(registration)
     can_manage_competition?(registration.competition) || (registration.user_id == self.id && editable_by_user)
   end
 
-  def can_delete_registration?(registration)
-    # A registration can only be deleted by a user after it has been accepted if the organizers allow
-    can_delete_by_user = (!registration.accepted? || registration.competition.registration_delete_after_acceptance_allowed?)
-    can_manage_competition?(registration.competition) || (registration.user_id == self.id && can_delete_by_user)
-  end
-
```

Comment:
Why not keep this method and use it in the registration checker, instead of having all incredibly dense logic in oh-so-many lines of code over there?

Reply:
Good suggestion! Implemented, but with the twist of putting it on the `registration` object which seems to make more sense in the new framework :D 

In general, there is a lot of scope to do more of this, but it's in #10864 as low priority given that it's more tech debt/refactoring work than critical to functionality - `registration_checker` (at least at the moment) is fairly maintainable